samples/qa-virtualgridlist/src/context/RecordContext.js

```diff
@@ -0,0 +1,168 @@
+import {createContext, useReducer} from "react";
+import {initializeRecords} from "../utils";
+
+export const RecordContext = createContext(null);
+export const RecordDispatchContext = createContext(null);
+
+export const RecordProvider = ({children}) => {
+	const initialData = initializeRecords();
+	const [state, dispatch] = useReducer(recordReducer, initialData);
+
+	return (
+		<RecordContext.Provider value={state}>
+			<RecordDispatchContext.Provider value={dispatch}>
+				{children}
+			</RecordDispatchContext.Provider>
+		</RecordContext.Provider>
+	);
+};
+
+/* action type */
+const ADD_ITEM = 'recordReducer/ADDITEM';
+const CHANGE_DATASIZE = 'recordReducer/CHANGEDATASIZE';
+const CHANGE_MIN_HEIGHT = 'recordReducer/CHANGEMINHEIGHT';
+const CHANGE_MIN_WIDTH = 'recordReducer/CHANGEMINWIDTH';
+const CHANGE_SPACING = 'recordReducer/CHANGESPACING';
+const DELETE_ITEM = 'recordReducer/DELETEITEM';
+const DELETE_SELECTED_ITEM = 'recordReducer/DELETESELECTEDITEM';
+const SELECT_ALL = 'recordReducer/SELECTALL';
+const SELECT_ITEM = 'recordReducer/SELECTITEM';
+const SELECTION_ENABLE = 'recordReducer/SELECTIONENABLE';
+const SET_DATA = 'recordReducer/SETDATA';
+
+const addItem = (item) => ({type: ADD_ITEM, item});
+const changeDataSize = (size) => ({type: CHANGE_DATASIZE, size});
+const changeMinHeight = (size) => ({type: CHANGE_MIN_HEIGHT, size});
+const changeMinWidth = (size) => ({type: CHANGE_MIN_WIDTH, size});
+const changeSpacing = (size) => ({type: CHANGE_SPACING, size});
+const deleteItem = () => ({type: DELETE_ITEM});
+const deleteSelectedItem = () => ({type: DELETE_SELECTED_ITEM});
+const selectAll = () => ({type: SELECT_ALL});
+const selectItem = (dataIndex) => ({type: SELECT_ITEM, dataIndex});
+const selectionEnable = () => ({type: SELECTION_ENABLE});
+const setData = (item, index) => ({type: SET_DATA, item, index});
+
+export {
+	addItem,
+	changeDataSize,
+	changeMinHeight,
+	changeMinWidth,
+	changeSpacing,
+	deleteItem,
+	deleteSelectedItem,
+	selectAll,
+	selectItem,
+	selectionEnable,
+	setData
+};
+
+export default function recordReducer (state, action) {
+	switch (action.type) {
+		case (ADD_ITEM): {
```

Comment:
I think this is not a matter of preference. An independent block scope must be created to prevent variables from overlapping.
(https://stackoverflow.com/questions/42480949/what-do-the-curly-braces-do-in-switch-statement-after-case-in-es6)

Reply:
I meant ( ) from ADD_ITEM :)

Reply:
Ok. I took it as a different situation at first. It's modified